src/config.rs

```diff
@@ -0,0 +1,198 @@
+    #[serde(
+        deserialize_with = "Config::deserialize_update_frequency",
+        default = "Config::default_update_frequency"
+    )]
+    pub update_frequency: chrono::Duration,
+}
+impl Config {
+    /// Custom deserialization for update frequency.
+    ///
+    /// We don't want to specify the duration in seconds, as that's annoying, so implementing
+    /// a shorthand notation.
+    fn deserialize_update_frequency<'de, D>(deserializer: D) -> Result<chrono::Duration, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        use serde::de::Error;
+        let source = String::deserialize(deserializer)?;
+        let regex = lazy_regex!("([[:digit:]]+) *([hmsd])");
+        let found = regex.captures(&source).ok_or_else(|| {
+            D::Error::invalid_value(
+                Unexpected::Str(&source),
+                &"numbers followed by a unit d/h/m/s",
+            )
+        })?;
+        let digits = found.get(1).expect("we should have digits");
+        let unit = found.get(2).expect("we should have a unit");
+        let digits: i64 = digits
+            .as_str()
+            .parse()
+            .map_err(|_| D::Error::invalid_value(Unexpected::Str(digits.as_str()), &"numbers"))?;
+        let unit: char = unit.as_str().parse().map_err(|_| {
+            D::Error::invalid_value(Unexpected::Str(unit.as_str()), &"a unit d/h/m/s")
+        })?;
+        let result = match unit {
+            'd' => chrono::Duration::days(digits),
+            'h' => chrono::Duration::hours(digits),
+            'm' => chrono::Duration::minutes(digits),
+            's' => chrono::Duration::seconds(digits),
+            _ => unreachable!(),
+        };
+        Ok(result)
+    }
+
+    fn default_update_frequency() -> chrono::Duration {
+        chrono::Duration::hours(2)
+    }
+}
+
+
+#[cfg(test)]
+mod test {
+    use crate::{config::RepoName, config::Config};
+
+    use super::ProjectToHook;
+
+    /// Can a typical config be parsed?
+    #[test]
+    fn test_config_parse() {
+        let source = r#"
+            projects:
+                - url: "https://github.com/owner1/project1"
+                - url: "https://github.com/owner2/project2"
+            update_frequency: 15m
+        "#;
+        let config: Config = serde_yaml::from_str(source).unwrap();
+        assert_eq!(config.update_frequency, chrono::Duration::minutes(15));
+        assert_eq!(config.projects.len(), 2);
+        assert_eq!(config.projects[0].owner, "owner1");
+        assert_eq!(config.projects[0].repo, RepoName::from("project1"));
+        assert_eq!(config.projects[1].owner, "owner2");
+        assert_eq!(config.projects[1].repo, RepoName::from("project2"));
+    }
+
+    /// Can a typical ProjectToHook be parsed?
+    #[test]
+    fn test_project_to_hook_parse() {
+        let source = "https://github.com/owner1/project1=https://hooks.slack.com/services/YOUR/SLACK/HOOK";
```

Comment:
Since it always expects a GitHub repo, wouldn't it be simple to store `{owner}/{project}`? Maybe the same for the Slack hook.
```
{owner}/{project}=YOUR/SLACK/HOOK
```

Reply:
Since keeping the entire URL isn't much of a bother and will simplify our life later if we ever decide to expand to GitLab, I think I prefer it as is.

WDYT?

Reply:
For the repos, I think it makes sense.

I was wondering if this tool would only be for open-source repos or also for internal ones.

Reply:
One of these days, I'll ask around if there's any interest for internal repos.